### PR TITLE
Add support for OpenTelemetry 0.16 and tracing-opentelemetry 0.15

### PR DESCRIPTION
**Pull request by wolverian**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
       - uses: actions-rs/cargo@v1
         with:
           command: test
-          args: --workspace --all-targets --features opentelemetry_0_15
+          args: --workspace --all-targets --features opentelemetry_0_16
```
Comment (LukeMathWalker) [inline .github/workflows/ci.yml:24]: It'd be ideal to run tests for each version of OTEL - can you amend?

Comment (wolverian) [reply .github/workflows/ci.yml]: I added a matrix to the jobs that used `--features`. Not sure it's really needed on the other jobs besides test, but they have it now.

Test run here: https://github.com/wolverian/reqwest-middleware/actions/runs/1181732588

Comment (LukeMathWalker) [reply .github/workflows/ci.yml]: Better safe than sorry, thanks!

**Comment by LukeMathWalker** [comment]
One last issue - can you update the README to include the newer feature flag?

**Comment by wolverian** [comment]
> One last issue - can you update the README to include the newer feature flag?

Of course! I added it where the other features are mentioned. I did not reformat the line because I'm not sure if there's a specific formatting style for Markdown you'd prefer.

**Review by LukeMathWalker — APPROVED** [review]

**Pull request merged**